codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/org/example/ktor/core/extension/HttpClientExt.kt

```diff
@@ -0,0 +1,22 @@
+package org.example.ktor.core.extension
+
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.defaultRequest
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.json.Json
+
+fun HttpClient.configuredForApi() = config {
```

Comment:
I think this function could both, create and configure an HttpClient, since you don't reuse this extension from tests:
```suggestion
fun createHttpClient() = HttpClient {
```

In real life this method would be called in some DI framework providing a singleton HttpClient.